h2o-algos/src/main/java/hex/util/LinearAlgebraUtils.java

```diff
@@ -233,31 +250,54 @@ else if (!_ainfo._catMissing[p])
    * @param transpose Should result be transposed to get L?
    * @return L or R matrix from Cholesky of Y Gram
    */
-  public static double[][] computeR(Key<Job> jobKey, DataInfo yinfo, boolean transpose) {
+  public static double[][] computeR(Key<Job> jobKey, DataInfo yinfo, boolean transpose, double[][] xx) {
     // Calculate Cholesky of Y Gram to get R' = L matrix
     Gram.GramTask gtsk = new Gram.GramTask(jobKey, yinfo);  // Gram is Y'Y/n where n = nrow(Y)
     gtsk.doAll(yinfo._adaptedFrame);
+    Matrix ygram = null;
     // Gram.Cholesky chol = gtsk._gram.cholesky(null);   // If Y'Y = LL' Cholesky, then R = L'
-    Matrix ygram = new Matrix(gtsk._gram.getXX());
-    CholeskyDecomposition chol = new CholeskyDecomposition(ygram);
+    if (xx==null)
+      ygram = new Matrix(gtsk._gram.getXX());
+    else
+      ygram = new Matrix(gtsk._gram.getXX(xx));
 
+    CholeskyDecomposition chol = new CholeskyDecomposition(ygram);
```

Comment:
Why don't we do gtsk._gram.computeCholesky() instead of using Jama? our Cholesky solver is multithreaded
and way faster than Jama.